csaf_2_1/recommendedTests/recommendedTest_6_2_44.js

```diff
@@ -0,0 +1,148 @@
+ * Check whether the license identifiers ref is a deprecated Aboutcode's license
+ * @param {string} licenseRefToCheck
+ * @return {boolean}
+ */
+function isDeprecatedLicenseRef(licenseRefToCheck) {
+  if (!licenseRefToCheck.startsWith(ABOUT_CODE_LICENSE_REF_PREFIX)) {
+    return false
+  } else {
+    const licenseKey = licenseRefToCheck.substring(
+      ABOUT_CODE_LICENSE_REF_PREFIX.length
+    )
+    return DEPRECATED_ABOUT_CODE_LICENSE_KEYS.has(licenseKey)
+  }
+}
+
+/**
+ * Recursively checks if a parsed license expression contains deprecated licenses
+ *
+ * @param {import('license-expressions').ParsedSpdxExpression} parsedExpression - The parsed license expression
+ * @returns {Array<string>} all deprecated licenses
+ */
+function allDeprecatedLicenses(parsedExpression) {
+  /** @type {Array<string>} */
+  const deprecatedLicenses = []
+  // If it's a LicenseRef type directly
+  if (
+    'licenseRef' in parsedExpression &&
+    isDeprecatedLicenseRef(parsedExpression.licenseRef)
+  ) {
+    deprecatedLicenses.push(parsedExpression.licenseRef)
+  }
+
+  if (
+    'license' in parsedExpression &&
+    DEPRECATED_SPDX_LICENSE_KEYS.has(parsedExpression.license)
+  ) {
+    deprecatedLicenses.push(parsedExpression.license)
+  }
+
+  if (
+    'exception' in parsedExpression &&
+    parsedExpression.exception &&
+    DEPRECATED_SPDX_LICENSE_KEYS.has(parsedExpression.exception)
+  ) {
+    deprecatedLicenses.push(parsedExpression.exception)
+  }
+
+  // If it's a conjunction, check both sides
+  if ('conjunction' in parsedExpression) {
+    deprecatedLicenses.push(...allDeprecatedLicenses(parsedExpression.left))
+    deprecatedLicenses.push(...allDeprecatedLicenses(parsedExpression.right))
+  }
+
+  // If it's a LicenseInfo type, it doesn't contain not listed licenses
+  return deprecatedLicenses
+}
+
+/**
+ * Checks if a license expression string contains deprecated licenses
+ *
+ * @param {string} licenseToCheck - The license expression to check
+ * @returns {Array<string>} all deprecated licenses
+ */
+export function allDeprecatedInLicenseString(licenseToCheck) {
+  const parseResult = parse(licenseToCheck)
+  return allDeprecatedLicenses(parseResult)
+}
+
+/**
+ * It MUST be tested that all license identifier and exceptions used are not deprecated.
+ * This SHALL be tested for the SPDX license list and Aboutcode's "ScanCode LicenseDB".
+ * The test MAY be skipped for other license inventorying entities.
+ *
+ * @param {unknown} doc
+ */
+export function recommendedTest_6_2_44(doc) {
+  /*
+        The `ctx` variable holds the state that is accumulated during the test run and is
+        finally returned by the function.
+       */
+  const ctx = {
+    warnings:
+      /** @type {Array<{ instancePath: string; message: string }>} */ ([]),
+  }
+
+  if (!validateSchema(doc)) {
+    return ctx
+  }
+
+  const licenseToCheck = doc.document.license_expression
+
+  if (validate(licenseToCheck).valid) {
+    const deprecatedLicenseIdentifier =
```

Comment:
Shouldn't that be plural as we are handling an array here?
```suggestion
    const deprecatedLicenseIdentifiers =
```